### PR TITLE
ES6 rewrite: Merge dev branch into master

### PR DESCRIPTION
**Pull request by mklabs**
This PR includes the rework of the lib into ES6 with babel configured to do so.

It cannot be merged automatically and still need work, namely a thorough code review and tests to ensure everything works as usual (and proper versionning, will probably bump to a major new 1.0.0 version to make sure nobody is impacted by this rather big rework).

It also includes some little changes needed by the CLI / dashboard, like "MSG" events handled by socket.io in the [dashboard](https://github.com/mklabs/tiny-lr-cli)

### CODE REVIEW

**Comment by mklabs** [comment]
cc @hemanth @shama @elwayman02 Hi guys, I may need help here. I don't know if you have the time to do so, but I'd really like if someone could test it and confirm that it's still working as intended.

That being said, I tested it rather extensively while developing the CLI part and dashboard and noticed it was working the same.

I'll keep the PR opened for the moment, but I'm keen on merging it into master and working further from there.

version has been bumped to 1.0.0 so it shouldn't impact existing integrations.

File: examples/express/app.js [hunk 1]
```diff
@@ -1,49 +1,49 @@
...
-
-app
-  .use(logger())
-  .use('/', express.static(path.join(__dirname)))
-  .use(tinylr.middleware({ app: app }));
+var fs      = require('fs');
+var path    = require('path');
+var express = require('express');
+var tinylr  = require('../..');
+var debug   = require('debug')('tinylr:server');
```
Comment (hemanth) [inline examples/express/app.js]: These can be `const`.

File: lib/client.js [hunk 2]
```diff
@@ -1,88 +1,94 @@
...
+const WebSocket = require('faye-websocket');
+const debug     = require('debug')('tinylr:client');
 
-var util      = require('util');
-var events    = require('events');
-var WebSocket = require('faye-websocket');
+let idCounter = 0;
 
-module.exports = Client;
+class Client extends events.EventEmitter {
```
Comment (hemanth) [inline lib/client.js]: Planning for `exports default` ?

File: lib/client.js [hunk 3]
```diff
@@ -1,88 +1,94 @@
...
 
-function Client(req, socket, head, options) {
-  options = this.options = options || {};
-  this.ws = new WebSocket(req, socket, head);
-  this.ws.onmessage = this.message.bind(this);
-  this.ws.onclose = this.close.bind(this);
-  this.id = this.uniqueId('ws');
-}
+  constructor (req, socket, head, options) {
```
Comment (hemanth) [inline lib/client.js]: `options={}`

File: test/client.js [hunk 4]
```diff
@@ -1,52 +1,46 @@
...
-      .expect(200, function() {});
-  });
-
-});
+
+var request   = require('supertest');
+var assert    = require('assert');
+var parse     = require('url').parse;
+var listen    = require('./helpers/listen');
+var WebSocket = require('faye-websocket').Client;
```
Comment (hemanth) [inline test/client.js]: `const` again, if you are babelifying it, we could use `imports` as well.

File: lib/client.js (hunk 2 above)
Comment (mklabs) [inline lib/client.js]: should be :)

File: test/client.js (hunk 4 above)
Comment (mklabs) [inline test/client.js]: tests have been left as is, i also need to rewrite them :( sure thing for imports I need to make this move

**Comment by mklabs** [comment]
@hemanth Thanks for review, still in the process of changing requires into imports now.

**Comment by hemanth** [comment]
3pzQ8ghJkggi4

**Pull request merged**